test/JDBC/input/charindex_replace_patindex-vu-prepare.sql

```diff
@@ -0,0 +1,96 @@
+-- TEST MVU FOR PATINDEX, CHARINDEX, REPLACE FOR AI COLLATION
+CREATE DATABASE babel_5144_db COLLATE bbf_unicode_cp1_ci_ai
+GO
+
+USE babel_5144_db
+GO
+
+CREATE PROCEDURE babel_5144_p (@src sys.NVARCHAR(100), @from sys.NVARCHAR(100), @to sys.NVARCHAR(100)) AS
+SELECT 'running replace'
+SELECT replace(@src, @from, @to);
+SELECT 'running charindex'
+SELECT charindex(@from, @src);
+SELECT 'running patindex'
+SELECT patindex('%de%', @src);
+GO
+
+CREATE FUNCTION babel_5144_f1(@src sys.NVARCHAR(100), @from sys.NVARCHAR(100), @to sys.NVARCHAR(100)) RETURNS NVARCHAR(100)
+AS
+BEGIN
+    RETURN replace(@src, @from, @to);
+END
+GO
+
+CREATE FUNCTION babel_5144_f2(@src sys.NVARCHAR(100), @ch sys.NVARCHAR(100)) RETURNS NVARCHAR(100)
+AS
+BEGIN
+    RETURN charindex(@ch, @src);
+END
+GO
+
+CREATE FUNCTION babel_5144_f3(@src sys.NVARCHAR(100)) RETURNS NVARCHAR(100)
+AS
+BEGIN
+    RETURN patindex('%de%', @src);
+END
+GO
+
+CREATE TABLE babel_5144_t1 (src NVARCHAR(100),
+                            substr1 NVARCHAR(100),
+                            substr2 NVARCHAR(100),
+                            [replaced] AS replace(src, substr1, substr2),
+                            [charIndex] as charindex(substr1, src),
+                            [patindex] AS patindex('%de%', src));
+GO
+
+CREATE TABLE babel_5144_t2 (src NVARCHAR(100),
+                            substr1 NVARCHAR(100),
+                            substr2 NVARCHAR(100),
+                            CHECK (replace(src, substr1, substr2) != src),
+                            CHECK (charindex(substr1, src) >= 4),
+                            CHECK (patindex('%de%', src) > 0));
+GO
+
+-- Enable all commente lines after BABEL-5155 is fixed
+-- CREATE TABLE babel_5144_t3 (src NVARCHAR(100),
+--                             substr1 NVARCHAR(100),
+--                             substr2 NVARCHAR(100),
+--                             [replaced] NVARCHAR(100) DEFAULT replace('abcḍèĎÈdedEDEabcd', 'de', '##'),
```

Comment:
What if we cast the input into varchar/nvarchar, how does that affect the output then?